### DISCUSSION TITLE
Efficiency regression on verifying non-linear arithmetic properties between 4.11.2 and 4.12.2

### DISCUSSION THREAD

**Discussion by rahxephon89** [Z3Prover/z3#6858] [Q&A] [answered]
Hi, we are observing a regression on verifying programs using non-linear arithmetic operations such as division and multiplication when switching from 4.11.2 to 4.12.2. I am wondering if there are some changes on options that may cause this? Thanks.

**Comment by NikolajBjorner** [comment]
try setting smt.arith.solver=2

**Reply by rahxephon89** [reply]
Thanks Nikola. Setting smt.arith.solver to 2 reduces the time but the verification fails. Is there anything else I can try?

**Reply by NikolajBjorner** [reply, marked as answer]
1. some improvements will go into the non-linear in the coming months so it could change the situation.
2. arith.solver=2 is incomplete so will often produce unknown for non-linear reals (and of course integers).

**Reply by rahxephon89** [reply]
Thanks Nikola!
